backend/canisters/group_index/impl/src/model/canisters_requiring_controller_swap.rs

```diff
@@ -0,0 +1,49 @@
+use candid::CandidType;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashSet, VecDeque};
+use tracing::info;
+use types::CanisterId;
+
+#[derive(CandidType, Serialize, Deserialize, Default)]
+pub struct CanistersRequiringControllerSwap {
```

Comment:
It would be good to expose the count pending / failed in the metrics so we can see the progress.